### PR TITLE
[Mistweaver] Add T30 Module

### PR DESCRIPTION
**Pull request by vohrr**
### Description
Add mistweaver's tier set module for the 10.1 raid
Does not need to wait to be merged since it wont be active unless you're wearing tier

### Screenshot
image

### Testing

Test Report: https://www.warcraftlogs.com/reports/Kpkfv92CYR8Vr6LG#type=healing&source=21&fight=9

### CODE REVIEW

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+              <strong>{formatNumber(this.vivHealing)}</strong> extra{' '}
+              <SpellLink id={SPELLS.VIVIFY.id} /> healing
+            </li>
+            <li>
+              <strong>{formatNumber(this.wastedManaGain)}</strong> mana wasted from overcapping
+            </li>
+          </ul>
+        }
+      >
+        <BoringValueText label="T29 Tier Set">
```
Comment (trevorm4) [inline src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]: T30? Alternatively you could name it after https://www.wowhead.com/ptr/item-set=1546/fangs-of-forged-vermillion

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  has4Piece: boolean = true;
+  vivHealing: number = 0;
+  renewingMistHealing: number = 0;
+  manaGain: number = 0;
+  wastedManaGain: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.has2Piece = this.selectedCombatant.has2PieceByTier(TIERS.T29);
+    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T29) && this.has2Piece;
```
Comment (trevorm4) [inline src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]:
```suggestion
    this.has2Piece = this.selectedCombatant.has2PieceByTier(TIERS.T30);
    this.has4Piece = this.selectedCombatant.has4PieceByTier(TIERS.T30) && this.has2Piece;
```

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        Events.heal.by(SELECTED_PLAYER).spell(FOUR_PIECE_SPELLS),
+        this.handle4PcHeal,
+      );
+    }
+  }
+
+  get total4PieceHealing() {
+    return this.renewingMistHealing + this.vivHealing;
+  }
+  get twoSetUptime() {
```
Comment (trevorm4) [inline src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]:
```suggestion
  }
  
  get twoSetUptime() {
```

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  handle4PcHeal(event: HealEvent) {
+    const spellId = event.ability.guid;
+    if (this.selectedCombatant.hasBuff(SPELLS.SOULFANG_VITALITY.id)) {
+      if (spellId === SPELLS.RENEWING_MIST_HEAL.id) {
+        this.renewingMistHealing += calculateEffectiveHealing(event, FOUR_PIECE_BONUS);
+      } else if (spellId === SPELLS.VIVIFY.id) {
+        this.vivHealing += calculateEffectiveHealing(event, FOUR_PIECE_BONUS);
+      }
+    }
+  }
```
Comment (trevorm4) [inline src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]:
```suggestion
  handle4PcHeal(event: HealEvent) {
    const spellId = event.ability.guid;
    if (this.selectedCombatant.hasBuff(SPELLS.SOULFANG_VITALITY.id)) {
     const increase = calculateEffectiveHealing(event, FOUR_PIECE_BONUS);
     spellId === SPELLS.RENEWING_MIST_HEAL.id ? this.renewingMistHealing += increase : this.vivifyHealing += increase;
    }
  }
```

Comment (trevorm4) [reply src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]: Not strictly necessary but I think this makes the code more concise.

**Review by trevorm4 — CHANGES_REQUESTED** [review]

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx (hunk 2 above)
Comment (vohrr) [reply src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]: oops

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx (hunk 4 above)
Comment (vohrr) [reply src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]: Agreed

File: src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx (hunk 1 above)
Comment (vohrr) [reply src/analysis/retail/monk/mistweaver/modules/dragonflight/tier/T30MWTier.tsx]: 🤦‍♂️

**Pull request merged**